lib/IDE/SwiftSourceDocInfo.cpp

```diff
@@ -746,6 +777,17 @@ struct RangeResolver::Implementation {
     return true;
   }
 
+  bool nodeContainSelection(ASTNode Node) {
+    // If the selection starts before the node, return false.
+    if (SM.isBeforeInBuffer(Start, Node.getStartLoc()))
+      return false;
+    // If the node ends before the selection, return false.
```

Comment:
Nitpick: these comments don't add much beyond what's in the source code; I'd just remove them.